app/src/main/java/com/android/feedme/model/data/CommentRepository.kt

```diff
@@ -29,15 +29,16 @@ class CommentRepository(private val db: FirebaseFirestore) {
   }
 
   /**
-   * Adds a comment to Firestore.
+   * Adds a comment to Firestore. Will OVERRIDE the comment Id and get a new from firestore.
    *
    * @param comment The Comment object to be added.
    * @param onSuccess Callback invoked on successful addition of the comment.
    * @param onFailure Callback invoked on failure to add the comment, with an exception.
    */
   fun addComment(comment: Comment, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection("comments")
-        .document(comment.authorId)
+    val newDocRef = db.collection("comments").document()
```

Comment:
Are we creating a collection for comments ? 
How will we link it to the corresponding recipe and profile, is it only through the userId and recipeId ?

Reply:
> Are we creating a collection for comments ?

No, here is the definition of `document()`  :
Returns a DocumentReference pointing to a new document with an auto-generated ID within this collection.

> How will we link it to the corresponding recipe and profile, is it only through the userId and recipeId ?

 If you look at the model, you will see that that for a recipe and a profile, each have a list containing commentId.

Reply:
The logic may be confusing,  first we create a comment with a dummy Id, then we ask firestore to give us a new id, and finally we update our comment with this id.

Reply:
Okay I see, I thought the a single comment was accessible 3 times in the database (user, recipe and comment), but since in the user and the recipe we only have a list of 'commentId' it should be fine :)